UserService/src/main/java/com/beaconfire/userservice/controller/UserAuthController.java

```diff
@@ -25,10 +26,12 @@
 public class UserAuthController {
 
     private final UserAuthService userAuthService;
+    private final EmailService emailService;
 
     @Autowired
-    public UserAuthController(UserAuthService userAuthService) {
+    public UserAuthController(UserAuthService userAuthService, EmailService emailService) {
         this.userAuthService = userAuthService;
+        this.emailService = emailService;
```

Comment:
Email Service should be inject as a feign client

Reply:
I am still confused about this part. In the handout, it says email service can only consume message from rabbitMQ. Therefore, I put the producer part (EmailService class) in the UserService. And those utils stuff are only used in the producer.

Reply:
For the EmailService as a microservice, it only handles the consumer logic.